chia/server/ws_connection.py

```diff
@@ -310,6 +329,89 @@ async def outbound_handler(self) -> None:
+                self.log.error(f"Non existing function: {message_type}")
+                raise ProtocolError(Err.INVALID_PROTOCOL_MESSAGE, [message_type])
+
+            metadata = get_metadata(function=f)
+            if metadata is None:
+                self.log.error(f"Peer trying to call non api function {message_type}")
+                raise ProtocolError(Err.INVALID_PROTOCOL_MESSAGE, [message_type])
+
+            # If api is not ready ignore the request
+            if hasattr(self.api, "api_ready"):
+                if self.api.api_ready is False:
+                    return None
+
+            timeout: Optional[int] = 600
+            if metadata.execute_task:
+                # Don't timeout on methods with execute_task decorator, these need to run fully
+                self.execute_tasks.add(task_id)
+                timeout = None
+
+            if metadata.peer_required:
+                coroutine = f(full_message.data, self)
+            else:
+                coroutine = f(full_message.data)
+
+            async def wrapped_coroutine() -> Optional[Message]:
+                try:
+                    # hinting Message here is compensating for difficulty around hinting of the callbacks
+                    result: Message = await coroutine
+                    return result
+                except asyncio.CancelledError:
+                    pass
+                except Exception as e:
+                    tb = traceback.format_exc()
+                    self.log.error(f"Exception: {e}, {self.get_peer_logging()}. {tb}")
+                    raise
+                return None
+
+            response: Optional[Message] = await asyncio.wait_for(wrapped_coroutine(), timeout=timeout)
+            self.log.debug(
+                f"Time taken to process {message_type} from {self.peer_node_id} is "
+                f"{time.time() - start_time} seconds"
+            )
+
+            if response is not None:
+                response_message = Message(response.type, full_message.id, response.data)
+                await self.send_message(response_message)
+        except TimeoutError:
+            self.log.error(f"Timeout error for: {message_type}")
+        except Exception as e:
+            if not self.closed:
+                tb = traceback.format_exc()
+                self.log.error(f"Exception: {e} {type(e)}, closing connection {self.get_peer_logging()}. {tb}")
+            else:
+                self.log.debug(f"Exception: {e} while closing connection")
+            # TODO: actually throw one of the errors from errors.py and pass this to close
+            await self.close(API_EXCEPTION_BAN_SECONDS, WSCloseCode.PROTOCOL_ERROR, Err.UNKNOWN)
+        finally:
+            if task_id in self.api_tasks:
+                self.api_tasks.pop(task_id)
+            if task_id in self.execute_tasks:
+                self.execute_tasks.remove(task_id)
+
+    async def incoming_message_handler(self) -> None:
+        while True:
```

Comment:
so, with this patch there is no limit to the number of tasks we create. Is that the case currently also, prior to this task? (if not, this would be a regression)

Reply:
There was no limit before. Tasks were just created out of the queue like they appeared.

Reply:
Resolving, as this appears to have been since approved.